cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,452 @@
+                    // to be saved with the content. We therefore need to filter out the attribute
+                    // upon serlialization (which happens on "save", view sourcecode and preview
+                    // among others).
+                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.serializer/#addnodefilter
+                    editor.serializer.addNodeFilter("conditional-block", (nodes) => {
+                        // Iterate through all filtered nodes and remove the contenteditable attribute
+                        // as well as the wrapper <div> we add when creating the custom element
+                        nodes.forEach((node) => {
+                            if (!!node.attr("contenteditable")) {
+                                // Remove the contenteditable attribute on the <conditional-block>
+                                node.attr("contenteditable", null);
+                                // Traverse into the first child which would be the <div contenteditable="true">
+                                // https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.node/#walk
+                                const content = node.walk();
+                                // Remove the div but keep the content
+                                // https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.node/#unwrap
+                                content.unwrap();
+                            }
+                        });
+                    });
+                });
+
+                // Register a custom toolbar menu button to insert the conditional block
+                // https://www.tiny.cloud/docs/ui-components/typesoftoolbarbuttons/#menubutton
+                editor.ui.registry.addButton("conditionalblock", {
+                    icon: "conditional-block",
+                    tooltip: "Insert conditional block",
+                    onAction: () => {
+                        dialogManager(null, editor);
+                    }
+                });
+            },
+
+            // The following css will be injected into the editor, extending
+            // the default styles.
+            // In a real world scenario, with much more custom styles for headings
+            // links, tables, images etc, it's recommended to use the content_css
+            // option to load a separate CSS file. Makes editing easier too.
+            // https://www.tiny.cloud/docs/configure/content-appearance/#content_style
+            // https://www.tiny.cloud/docs/configure/content-appearance/#content_css
+            content_style: `
+                /* We remove the default selected outline because it doesn't follow the
+                 * border radius we have on the conditional-block element and move it to the
+                 * conditional-block element
+                 * /
+                .mce-content-body conditional-block[contentEditable=false][data-mce-selected] {
```

Comment:
Is the `contenteditable=false` selector really needed here? As aren't all of these contenteditable false? Also I'm reasonably sure you can't get the `data-mce-selected` attribute normally for custom elements.